### ISSUE TITLE
DepthRenderer method='median' is not differentiable

### ISSUE THREAD

**Issue by gtangg12** [nerfstudio-project/nerfstudio#3334]
https://github.com/nerfstudio-project/nerfstudio/blob/702886b213a5942b64250b3c4dc632e62237a511/nerfstudio/model_components/renderers.py#L330

Seems like it will create a lot of "why is this not learning issues" downstream.

**Comment by peasant98** [comment]
I agree, it might be good to have the method as "expected" by default

There was a small bug where expected depth wasn't previously used in `depth-nerfacto`:

https://github.com/nerfstudio-project/nerfstudio/pull/2852/files
